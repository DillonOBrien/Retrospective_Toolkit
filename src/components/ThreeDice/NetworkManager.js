/* eslint-disable no-console */
import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import * as io from "socket.io-client";
import DiceLanding from "../../containers/DiceLanding";

const NetworkManager = () => {
  const [socket, setSocket] = useState();
  const { roomId } = useParams();
  console.log("rendering");

  // check with server to see if roomId exists
  // if () === null) {};
  // redirect to page not found

  useEffect(() => {
    setSocket(io(window.location.origin.replace(window.location.port, "5000")));
  }, []);

  // eslint-disable-next-line consistent-return
  useEffect(() => {
    if (socket) {
      console.log("on fired");
      socket.emit("board:create", roomId);
      console.log("emit fired");
      socket.on("create:complete", (msg) => {
        console.log(msg);
      });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [socket]);

  return <>{socket && <DiceLanding socket={socket} roomId={roomId} />}</>;
};

export default NetworkManager;
