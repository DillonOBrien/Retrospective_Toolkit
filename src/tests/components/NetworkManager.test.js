import React from "react";
import { shallow } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import NetworkManager from "../../components/ThreeDice/NetworkManager";

let wrapper;

beforeEach(() => {
  wrapper = shallow(
    <MemoryRouter initialEntries={["/retro/room:Id"]} keyLength={0}>
      <NetworkManager />
    </MemoryRouter>
  ).dive();
});

describe("ThreeDice/NetworkManager Testing", () => {
  // Snapshot for NetworkManager
  test("SNAPSHOT: NetworkManager", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
