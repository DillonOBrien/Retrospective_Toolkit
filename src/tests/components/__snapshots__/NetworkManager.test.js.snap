// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreeDice/NetworkManager Testing SNAPSHOT: NetworkManager 1`] = `
<ContextProvider
  value={
    Object {
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/retro/room:Id",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/retro/room:Id",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "location": Object {
        "hash": "",
        "pathname": "/retro/room:Id",
        "search": "",
        "state": undefined,
      },
      "match": Object {
        "isExact": false,
        "params": Object {},
        "path": "/",
        "url": "/",
      },
      "staticContext": undefined,
    }
  }
>
  <ContextProvider
    value={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/retro/room:Id",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/retro/room:Id",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NetworkManager />
  </ContextProvider>
</ContextProvider>
`;
